modules/apps/transfer/client/cli/tx.go

```diff
@@ -85,24 +86,51 @@ corresponding to the counterparty channel. Any timeout set to 0 is disabled.`),
 			// if the timeouts are not absolute, retrieve latest block height and block timestamp
 			// for the consensus state connected to the destination port/channel
 			if !absoluteTimeouts {
-				consensusState, height, _, err := channelutils.QueryLatestConsensusState(clientCtx, srcPort, srcChannel)
+				clientRes, err := channelutils.QueryChannelClientState(clientCtx, srcPort, srcChannel, false)
 				if err != nil {
 					return err
 				}
 
+				var clientState exported.ClientState
+				if err := clientCtx.InterfaceRegistry.UnpackAny(clientRes.IdentifiedClientState.ClientState, &clientState); err != nil {
+					return err
+				}
+
+				clientHeight, ok := clientState.GetLatestHeight().(clienttypes.Height)
+				if !ok {
+					return fmt.Errorf("invalid height type. expected type: %T, got: %T", clienttypes.Height{}, clientState.GetLatestHeight())
+				}
+
+				var consensusState exported.ConsensusState
+				if clientState.ClientType() != exported.Localhost {
```

Comment:
I wonder if this command works for solo machines?

Reply:
> I wonder if this command works for solo machines?

I think it should. 

> Could we avoid checking explicitly for Localhost here? 

We could and just use local clock time as the standard. But I'm happy to defer that til later, I'll open an issue for it. 
But I think we should just proceed with this approach and backport for 7.1. I don't see many future client implementations not having the concept of consensus states, I think localhost is an outlier

Reply:
Solo machines don't set consensus states so it shouldn't work? This appears to be true already so happy to see it fixed in #3594